### PR TITLE
[POAE7-2399] [M2 Integration] velox to substrait join transform support 

### PR DESCRIPTION
**Pull request by chaojun-zhang**
### What changes were proposed in this pull request?

 velox to substrait join transform support & add inner join test cases

### Why are the changes needed?

 velox to substrait join transform support 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT
### Which label does this PR belong to?

veloxIntegration,test

### CODE REVIEW

File: cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
  */
 
 #include "VeloxPlanFragmentToSubstraitPlan.h"
+#include <google/protobuf/util/json_util.h>
```
Comment (ZJie1) [inline cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp:23]: still need this now?

File: cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp [hunk 2]
```diff
@@ -109,13 +108,47 @@ void VeloxPlanFragmentToSubstraitPlan::reconstructVeloxPlan(
       planBuilder_->addNode([&](std::string id, core::PlanNodePtr input) {
         return std::make_shared<ValuesNode>(valuesNode->id(), valuesNode->values());
       });
+    } else if (auto joinNode =
+                   std::dynamic_pointer_cast<const AbstractJoinNode>(*riter)) {
+      const auto& joinLeftSource = joinNode->sources()[0];
+      const auto& joinRightSource = joinNode->sources()[1];
+
+      auto leftValuesNode = std::make_shared<ValuesNode>(
```
Comment (ZJie1) [inline cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp:116]: What about the case if there not only have one join node ,such as , it's filter->project->join?

Comment (chaojun-zhang) [reply cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp:116]: not support now , consider it as TODO item

File: cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp (hunk 1 above)
Comment (chaojun-zhang) [reply cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp]: Fixed

**Review by ZJie1 — APPROVED** [review]

**Review by yeyuqiang — APPROVED** [review]

**Review by harborn — APPROVED** [review]

**Pull request merged**